WordPress/openverse-catalog
# Update Smithsonian Unit code checker DAG to alert to Slack

## Fixes


Fixes #420

## Description


On main, we have a DAG called `check_new_smithsonian_unit_codes_workflow` which can only be run manually (it's not on a schedule), which compares the configuration in the `SMITHSONIAN_SUB_PROVIDERS` dictionary with the unit codes available in the Smithsonian API and raises an exception if they have fallen out of sync. It also creates and populates a DB table with the codes that need to be synced, which appears to be for legacy purposes and is no longer used anywhere.

This PR:

* Removes the two outdated unit codes from the Smithsonian sub provider configuration
* Entirely removes the `check_new_smithsonian_unit_codes_workflow` DAG and moves the logic into the Smithsonian provider script itself, so that it will fail fast if the unit code configuration needs to be changed.
* Removes logic for creating/updating DB table and instead reports the needed changes to Slack in a message like this:

<img width="644" alt="Screen Shot 2022-04-04 at 3 15 24 PM" src="https://user-images.githubusercontent.com/63313398/161640943-0eb8cf21-6d64-4c1c-a631-0fc01de5f66e.png">

## Notes
* Currently if the unit code validation fails, we'll get the Slack message with the error, but the `load_complete` Slack message will _also_ fire. There's an issue to address that: #465


## Testing Instructions

**Prereqs to test Slack:**
* You'll need to make sure you have the Slack connection set up for the Alerts channel. In your .env, that's `AIRFLOW_CONN_SLACK_ALERTS=<slack_connection_url>`. Troubleshooting tip: make sure it is formatted correctly with double-prepended 'https' and URL encoding (eg `"https://https%3A%2F%2Fhooks.slack.com%2Fservices%2everythingelse"`)
* You'll need to override the default behavior of omitting Slack messages locally. Go into the Airflow UI and select Admin > Variables, then add a variable named `slack_message_override` with value `true`.

Test:
* Check out this branch and run the Smithsonian provider script. With the changes in this PR, it should run successfully.
    * Watch the logs for the `pull_data` task and verify it starts writing records to disk, then mark the task as successful.
* Manually update the `SMITHSONIAN_SUB_PROVIDERS` dictionary in `provider_details.py` to make the unit codes out of sync. Try adding in non-existing codes ("FOO" etc), and deleting existing codes.
    * Run the provider script again and verify that the `pull_data` task fails and the error is sent to Slack. 

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
